### PR TITLE
Fixes NPE in jUnit tests

### PR DESCRIPTION
**Pull request by dbacinski**
@RxLog* annotations are not preserved in jUnit tests which cases NPE

Changed retention policy to RUNTIME.

### CODE REVIEW

**Comment by android10** [comment]
Make sure you are using these guidelines:
https://github.com/android10/java-code-styles

File: frodo-android-sample/src/test/java/com/fernandocejas/example/frodo/sample/ObservableSampleTest.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import rx.Observable;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(RobolectricTestRunner.class)
+public class ObservableSampleTest {
+
+    @Test
+    public void returnsObservableInteger() throws Exception {
+        //given
```
Comment (android10) [inline frodo-android-sample/src/test/java/com/fernandocejas/example/frodo/sample/ObservableSampleTest.java]: It is not necessary to add comments for BDD. This generates noise and should be implicit.

Comment (dbacinski) [inline frodo-android-sample/src/test/java/com/fernandocejas/example/frodo/sample/ObservableSampleTest.java]: I am following Spock syntax with BDD labels.

You're the boss. I will remove comments.

**Comment by dbacinski** [comment]
@android10 
https://github.com/android10/java-code-styles fork is 25 commit behind Square version. 
Wouldn't it be better to use Square version?

File: frodo-android-sample/src/test/java/com/fernandocejas/example/frodo/sample/ObservableSampleTest.java (hunk 1 above)
Comment (android10) [inline frodo-android-sample/src/test/java/com/fernandocejas/example/frodo/sample/ObservableSampleTest.java]: haha I'm not. It is just for consistency with the actual code base :)

**Comment by android10** [comment]
@dbacinski I will update my checkstyle repo. Thanks :smile:

**Comment by android10** [comment]
@dbacinski by the way, can you squash all the commits into one? So we don't pollute git history :smile:

**Comment by dbacinski** [comment]
Done.

File: frodo-android-sample/src/test/java/com/fernandocejas/example/frodo/sample/ObservableSampleTest.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package com.fernandocejas.example.frodo.sample;
```
Comment (android10) [inline frodo-android-sample/src/test/java/com/fernandocejas/example/frodo/sample/ObservableSampleTest.java]: I do not think we need this test for the sample. The main reason is that the samples are likely to be changed and this is only testing what the observable emits which makes our code very fragile to change. 
By the way, tests means more code to maintain and in this case, what is important is the domain of the library. 

Can you remove this test and test dependencies above in ```build.gradle``` so I can merge your fix?

Comment (dbacinski) [inline frodo-android-sample/src/test/java/com/fernandocejas/example/frodo/sample/ObservableSampleTest.java]: done

**Comment by android10** [comment]
@dbacinski thanks so much for the contribution! :+1:

**Pull request merged**